nevergrad/functions/base.py

```diff
@@ -234,6 +233,60 @@ def update_leaderboard(identifier: str, loss: float, array: np.ndarray, verbose:
         pass  # better avoir bugs for this
 
 
+class ArrayExperimentFunction(ExperimentFunction):
+    """Combines a function and its parametrization for running experiments (see benchmark subpackage).
+    Extends ExperimentFunction, in the special case of an array, by allowing the creation of symmetries
+    of a single function. We can create ArrayExperimentFunction(callable, symmetry=i) for i in range(0, 2**d)
+    when the callable works on R^d.
+    Works only if there are no constraints.
+
+    Parameters
+    ----------
+    function: callable
+        the callable to convert
+    parametrization: Parameter
+        the parametrization of the function
+    symmetry: int
+        number parametrizing how we symmetrize the function.
+    """
+
+    def __init__(self, function: tp.Callable[..., tp.Loss], parametrization: p.Parameter, symmetry: int = 0) -> None:
+        """Adds a "symmetry" parameter, which allows the creation of many symmetries of a given function.
+
+        symmetry: an int, 0 by default.
+        if not zero, a symmetrization is applied to the input; each of the 2^d possible values
+        for symmetry % 2^d gives one different function.
+        Makes sense if and only if (1) the input is a single ndarray (2) the domains are symmetric."""
+        self._inner_function = function
+        super().__init__(self.symmetrized_function, parametrization)
+        assert isinstance(parametrization, p.Array), f"{type(parametrization)} is not p.Array; {parametrization.parameters}."
+        assert (parametrization.bounds[0] is None) == (parametrization.bounds[1] is None)
+        assert len(parametrization._constraint_checkers) == 0
+        assert symmetry >= 0
+        assert symmetry < 2 ** self.dimension
```

Comment:
if you plan to apply this on experiment functions, maybe you'd need to extract its descriptors? or it's not a plan?

Reply:
maybe it's not a good idea at all

Reply:
For the moment we have no constructor based on ExperimentFunction.